addons/website_sale/static/src/js/extra_info.js

```diff
@@ -1,76 +1,45 @@
-import publicWidget from "@web/legacy/js/public/public_widget";
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
 
-publicWidget.registry.websiteSaleExtraInfo = publicWidget.Widget.extend({
-    // /shop/extra_info
-    selector: '#class_extra_info',
 
-    /**
-     * @constructor
-     */
-    init: function () {
-        this._super.apply(this, arguments);
-
-        this.http = this.bindService('http');
-
-        this.submitButton =  document.getElementsByName('website_sale_main_button')[0]
-        this._boundSaveExtraInfo = this._onSaveExtraInfo.bind(this);
-        this.submitButton.addEventListener("click", this._boundSaveExtraInfo);
-        this.extraForm = document.querySelector('form.form_extra_info');
-    },
-
-    destroy() {
-        this.submitButton.removeEventListener("click", this._boundSaveExtraInfo);
-        this._super(...arguments);
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
+export class WebsiteSaleExtraInfo extends Interaction {
+    static selector = "#class_extra_info";
+    dynamicSelectors = {
+        _submitbutton: () => document.querySelector('[name="website_sale_main_button"]'),
+    };
```

Comment:
We want to keep the existing `dynamicSelectors` :)
```suggestion
    dynamicSelectors = {
        ...this.dynamicSelectors,
        _submitbutton: () => document.querySelector('[name="website_sale_main_button"]'),
    };
```